src/probnum/quad/_bayesquad.py

```diff
@@ -284,6 +283,143 @@ def bayesquad_from_data(
     return integral_belief, info
 
 
+def bayesquad_multilevel(
+    nodes: Tuple[np.ndarray, ...],
+    fun_diff_evals: Tuple[np.ndarray, ...],
+    kernels: Optional[Tuple[Kernel, ...]] = None,
+    measure: Optional[IntegrationMeasure] = None,
+    domain: Optional[DomainLike] = None,
+    options: Optional[dict] = None,
+) -> Tuple[Normal, Tuple[BQIterInfo, ...]]:
+    r"""Infer the value of an integral from given sets of nodes and function
+    evaluations using a multilevel method.
+
+    In multilevel Bayesian quadrature, the integral :math:`\int_\Omega f(x) d \mu(x)`
+    is (approximately) decomposed as a telescoping sum over :math:`L+1` levels:
+
+    .. math:: \int_\Omega f(x) d \mu(x) \approx :math:`\int_\Omega f_0(x) d
+    \mu(x)` + \sum_{l=1}^L :math:`\int_\Omega [f_l(x) - f_{l-1}(x)] d \mu(x)`,
+
+    where :math:`f_l` is an increasingly accurate but also increasingly expensive
+    approximation to :math:`f`. Bayesian quadrature is subsequently applied to
+    independently infer each of the :math:`L+1` integrals and the outputs are summed
+    to infer :math:`\int_\Omega f(x) d \mu(x)`.
+
+    Parameters
+    ----------
+    nodes
```

Comment:
Are there any requirements? E.g., do the nodes need to be nested?

Reply:
No, there are no such requirements. The only requirement is that the number of nodes for a given level obviously needs to match the number of functions evaluations provided for that level.